pkg/installation/service.go

```diff
@@ -0,0 +1,159 @@
+// }
+
+func (s *installationService) TriggerInstallation(kubeconfig *rest.Config, tillerYaml string, installerYaml string, configuration installation.Configuration) error {
+	return s.triggerAction(tillerYaml, installerYaml, configuration, s.kymaInstaller, s.kymaInstaller.PrepareInstallation, installAction)
+}
+
+func (s *installationService) TriggerUpgrade(kubeconfig *rest.Config, tillerYaml string, installerYaml string, configuration installation.Configuration) error {
+	return s.triggerAction(tillerYaml, installerYaml, configuration, s.kymaInstaller, s.kymaInstaller.PrepareUpgrade, upgradeAction)
+}
+
+func (s *installationService) triggerAction(
+	tillerYaml string,
+	installerYaml string,
+	configuration installation.Configuration,
+	installer installation.Installer,
+	prepareFunction func(installation.Installation) error,
+	actionName string) error {
+
+	installationConfig := installation.Installation{
+		TillerYaml:    tillerYaml,
+		InstallerYaml: installerYaml,
+		Configuration: configuration,
+	}
+
+	err := prepareFunction(installationConfig)
+	if err != nil {
+		return pkgErrors.Wrap(err, fmt.Sprintf("Failed to prepare %s", actionName))
+	}
+
+	installationCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// We are not waiting for events, just triggering installation
+	_, _, err = installer.StartInstallation(installationCtx)
+	if err != nil {
+		return pkgErrors.Wrap(err, fmt.Sprintf("Failed to start Kyma %s", actionName))
+	}
+
+	return nil
+}
+
+func (s *installationService) InstallKyma(kubeconfig *rest.Config, tillerYaml string, installerYaml string, configuration installation.Configuration, currentStep step.Step) error {
+	installationConfig := installation.Installation{
+		TillerYaml:    tillerYaml,
+		InstallerYaml: installerYaml,
+		Configuration: configuration,
+	}
+
+	err := s.kymaInstaller.PrepareInstallation(installationConfig)
+	if err != nil {
+		return pkgErrors.Wrap(err, "Failed to prepare installation")
+	}
+
+	installationCtx, cancel := context.WithTimeout(context.Background(), s.kymaInstallationTimeout)
+	defer cancel()
+
+	stateChannel, errChannel, err := s.kymaInstaller.StartInstallation(installationCtx)
+	if err != nil {
+		return pkgErrors.Wrap(err, "Failed to start Kyma installation")
+	}
+
+	err = s.waitForInstallation(currentStep, stateChannel, errChannel)
+	if err != nil {
+		return pkgErrors.Wrap(err, "Error while waiting for Kyma to install")
+	}
+
+	return nil
+}
+
+func (s *installationService) CheckInstallationState(kubeconfig *rest.Config) (installation.InstallationState, error) {
+	return installation.CheckInstallationState(kubeconfig)
+}
+
+func (s *installationService) TriggerUninstall(kubeconfig *rest.Config) error {
+	return installation.TriggerUninstall(kubeconfig)
+}
+
+func (s *installationService) waitForInstallation(currentStep step.Step, stateChannel <-chan installation.InstallationState, errorChannel <-chan error) error {
+	for {
```

Comment:
How do we know here when we get a timeout from the context?
And how do we print that out to the user?